### PR TITLE
fix: runner/module resolution logic

### LINKED ISSUE #561: Fails to resolve the test runner module
**Describe the bug**

I was updating libraries that used https://github.com/hugomrdias/playwright-test/pull/548 to currently released playwright, but sadly it seems to fail to resolve the module

```
playwright-test test --runner entail test/*.js

Error: Cannot resolve module "entail" from "/Users/gozala/Projects/entail"
    at resolveModule (file:///Users/gozala/Projects/entail/node_modules/.pnpm/playwright-test@11.0.2/node_modules/playwright-test/src/utils/index.js:524:11)
    at async resolveTestRunner (file:///Users/gozala/Projects/entail/node_modules/.pnpm/playwright-test@11.0.2/node_modules/playwright-test/src/utils/index.js:544:18)
    at async file:///Users/gozala/Projects/entail/node_modules/.pnpm/playwright-test@11.0.2/node_modules/playwright-test/cli.js:273:34
 ELIFECYCLE  Command failed with exit code 1.
```

**To Reproduce**

Execute above command

**Expected behavior**

I was expecting that module resolution would succeed

**Additional context**

I am not yet sure why that error is happening, but I suspect that module resolution logic provided in my PR and one that was shipped are different, later failing to account for the context in which resolution takes place.

I will investigate and provide more details as I find them.

### PR DESCRIPTION
**Pull request by Gozala**
fixes #561

### CODE REVIEW

File: src/utils/index.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ import ora from 'ora'
 import { createServer } from 'http'
 import polka from 'polka'
 import { createRequire } from 'module'
-import { fileURLToPath, pathToFileURL } from 'url'
+import { fileURLToPath } from 'url'
```
Comment (Gozala) [inline src/utils/index.js:17]:
```suggestion
import { fileURLToPath, pathToFileURL } from 'url'
```

File: src/utils/index.js [hunk 2]
```diff
@@ -506,25 +506,14 @@ export async function createCov(runner, coverage, file, outputDir) {
...
-
-  if (!out) {
-    try {
-      const filePath = path.resolve(base, id)
-      fs.accessSync(filePath, fs.constants.R_OK)
-      out = await import(pathToFileURL(filePath).toString())
-    } catch {}
+    // Note we need to ensure base has trailing `/` or the the
+    // last entry is gonig to be dropped during resolution.
+    return await import(createRequire(toDirectoryPath(base)).resolve(id))
```
Comment (Gozala) [inline src/utils/index.js:513]:
```suggestion
    const path = createRequire(toDirectoryPath(base)).resolve(id)
    const url = pathToFileURL(path)
    return await import(url)
```

File: src/utils/index.js [hunk 3]
```diff
@@ -506,25 +506,16 @@ export async function createCov(runner, coverage, file, outputDir) {
...
-    try {
-      const filePath = path.resolve(base, id)
-      fs.accessSync(filePath, fs.constants.R_OK)
-      out = await import(pathToFileURL(filePath).toString())
-    } catch {}
+    // Note we need to ensure base has trailing `/` or the the
+    // last entry is gonig to be dropped during resolution.
+    const path = createRequire(toDirectoryPath(base)).resolve(id)
+    const url = pathToFileURL(path)
+    return await import(url)
```
Comment (Gozala) [inline src/utils/index.js:515]:
```suggestion
    return await import(url.href)
```

**Comment by Gozala** [comment]
cc @hugomrdias as I'm unable to add you as a reviewer

**Review by hugomrdias — CHANGES_REQUESTED** [review]
theres 2 path here, either a module id or a path (relative or absolute) you version does not import from a module export path ie. <module/id>/<export>

in you case it seems you are in the test runner repo, so you need to use path version because `entail` is not in node_modules.

**Comment by Gozala** [comment]
> in you case it seems you are in the test runner repo, so you need to use path version because `entail` is not in node_modules.

I am pretty sure that is not the problem (I can verify if you want me to), problem is that on line 512 import will be resolved from `playwright-test/src/utils/index.js` path not from my package path and depending on where playwright-test is that may fail. E.g. if I do `npx playwright-test` it will end up in a entire different path than my package or it's dependencies

https://github.com/hugomrdias/playwright-test/blob/2d1c32deef36a80a1cf2bff13cc0617b477a6a9c/src/utils/index.js#L509-L528

**Comment by Gozala** [comment]
Here is the example illustrating my point above

```
pnpm test:web                                              ✭

> sync-multihash-sha2@1.0.0 test:web /Users/gozala/Projects/sha256-multihash
> npx playwright-test test --runner entail test/*.js --cov && nyc report

Error: Cannot resolve module "entail" from "/Users/gozala/Projects/sha256-multihash"
    at resolveModule (file:///Users/gozala/.npm/_npx/2c2b474230005785/node_modules/playwright-test/src/utils/index.js:524:11)
    at async resolveTestRunner (file:///Users/gozala/.npm/_npx/2c2b474230005785/node_modules/playwright-test/src/utils/index.js:544:18)
    at async file:///Users/gozala/.npm/_npx/2c2b474230005785/node_modules/playwright-test/cli.js:273:34
 ELIFECYCLE  Command failed with exit code 1.
```

You can find source here https://github.com/web3-storage/sync-multihash-sha2/pull/3

File: src/utils/index.js [hunk 4]
```diff
@@ -506,25 +506,16 @@ export async function createCov(runner, coverage, file, outputDir) {
...
-  } catch {}
-
-  if (!out) {
-    try {
-      const filePath = path.resolve(base, id)
-      fs.accessSync(filePath, fs.constants.R_OK)
-      out = await import(pathToFileURL(filePath).toString())
-    } catch {}
+    // Note we need to ensure base has trailing `/` or the the
+    // last entry is gonig to be dropped during resolution.
```
Comment (hugomrdias) [inline src/utils/index.js:512]:
```suggestion
    // last entry is going to be dropped during resolution.
```

**Review by hugomrdias — APPROVED** [review]

**Pull request merged**